### PR TITLE
Add smoke API test script for SetGlobalProperties

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Added initial test script for testing SetGlobalProperties happy path and also added new common module for smoke API testing.

`"GitHub issue link" ~ " https://github.com/SmartDeviceLink/sdl_atf/issues/97"`

### CODE REVIEW

File: test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		keyboardLayout = "QWERTY",
+		keypressMode = "SINGLE_KEYPRESS",
+		limitedCharacterList = {"a"},
+		language = "EN-US",
+		autoCompleteText = "Daemon, Freedom"
+	}
+}
+
+--[[ Local Functions ]]
+local function put_file(fileName, self)
```
Comment (Itileda) [inline test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua]: @AKalinich-Luxoft , suggest to move putFile to common.
Smoke suppose basic checks of all RPCs with all possible and valid parameters, so in other scripts PutFile RPC will be used as precondition before sending tested API.

File: test_scripts/Smoke/commonSmokeApi.lua [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        end)
+    end)
+end
+
+function commonSmokeApi.postconditions()
+  StopSDL()
+end
+
+return commonSmokeApi
+
```
Comment (Itileda) [inline test_scripts/Smoke/commonSmokeApi.lua]: @AKalinich-Luxoft , please remove redundant new line

File: test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua (hunk 1 above)
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua]: @Itileda fixed in 5320c5539bfe24a781724ca9cf32e4ea14839897

File: test_scripts/Smoke/commonSmokeApi.lua (hunk 2 above)
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/commonSmokeApi.lua]: @Itileda fixed in 5320c5539bfe24a781724ca9cf32e4ea14839897

File: test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+	end)
+
+	EXPECT_HMICALL("TTS.SetGlobalProperties", params.responseTtsParams)
+	:Do(function(_,data)
+		self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+	end)
+
+	self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+
```
Comment (AByzhynar) [inline test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua]: @AKalinich-Luxoft Redundant empty line

File: test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+-- Steps:
+-- appID requests SetGlobalproperties with timeoutPrompt, helpPrompt and other valid parameters
+
+-- Expected:
+
+-- SDL validates parameters of the request
+-- SDL checks if UI interface is available on HMI
+-- SDL checks if TTS interface is available on HMI
+-- SDL checks if SetGlobalproperties is allowed by Policies
+-- SDL checks if all parameters is allowed by Policies
```
Comment (AByzhynar) [inline test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua]: @AKalinich-Luxoft parameters **are**

File: test_scripts/Smoke/commonSmokeApi.lua [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+end
+
+function commonSmokeApi.start(pHMIParams, self)
+  self, pHMIParams = commonSmokeApi.getSelfAndParams(pHMIParams, self)
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self)
+  :Do(function()
+      self:initHMI(self)
+      :Do(function()
+          commonFunctions:userPrint(35, "HMI initialized")
```
Comment (AByzhynar) [inline test_scripts/Smoke/commonSmokeApi.lua]: @AKalinich-Luxoft Use user_modules/consts for color

File: test_scripts/Smoke/commonSmokeApi.lua [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+  self, pAppId, pUpdateFunction = commonSmokeApi.getSelfAndParams(pAppId, pUpdateFunction, self)
+  if not pAppId then pAppId = 1 end
+  self["mobileSession" .. pAppId] = mobile_session.MobileSession(self, self.mobileConnection)
+  self["mobileSession" .. pAppId]:StartService(7)
+  :Do(function()
+      local corId = self["mobileSession" .. pAppId]:SendRPC("RegisterAppInterface",
+        config["application" .. pAppId].registerAppInterfaceParams)
+      EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+        { application = { appName = config["application" .. pAppId].registerAppInterfaceParams.appName } })
+      :Do(function(_, d1)
```
Comment (AByzhynar) [inline test_scripts/Smoke/commonSmokeApi.lua]: @AKalinich-Luxoft Bad names `d1`, `d2`, etc

File: test_scripts/Smoke/commonSmokeApi.lua [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+    mobileSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+end
+
+
+function commonSmokeApi.postconditions()
+  StopSDL()
+end
+
+return commonSmokeApi
```
Comment (AByzhynar) [inline test_scripts/Smoke/commonSmokeApi.lua]: @AKalinich-Luxoft 
Please add an empty line here

File: test_scripts/Smoke/commonSmokeApi.lua (hunk 5 above)
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/commonSmokeApi.lua]: @AByzhynar thanks, will use it. See https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/1f9a96a07f40b1051bd1cd0e5a3b77a979c40def

File: test_scripts/Smoke/commonSmokeApi.lua (hunk 6 above)
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/commonSmokeApi.lua]: @AByzhynar renamed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/1f9a96a07f40b1051bd1cd0e5a3b77a979c40def

File: test_scripts/Smoke/commonSmokeApi.lua (hunk 7 above)
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/commonSmokeApi.lua]: @AByzhynar fixed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/1f9a96a07f40b1051bd1cd0e5a3b77a979c40def

File: test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua (hunk 4 above)
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua]: @AByzhynar fixed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/1f9a96a07f40b1051bd1cd0e5a3b77a979c40def

File: test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua (hunk 3 above)
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/API/001_SetGlobalProperties_PositiveCase_SUCCESS.lua]: @AByzhynar fixed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/1f9a96a07f40b1051bd1cd0e5a3b77a979c40def

File: test_scripts/Smoke/commonSmokeApi.lua [hunk 8]
```diff
@@ -0,0 +1,268 @@
+---------------------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/commonSmokeApi.lua]: @dboltovskyi done in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/f2c903293464469ebdf036ca148182d0c2bb5501

File: test_scripts/Smoke/commonSmokeApi.lua [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+  tbl.policy_table.consumer_friendly_messages.messages = nil
+  tbl.policy_table.device_data = nil
+  tbl.policy_table.module_meta = nil
+  tbl.policy_table.usage_and_error_counts = nil
+  tbl.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  tbl.policy_table.module_config.preloaded_pt = nil
+  tbl.policy_table.module_config.preloaded_date = nil
+end
+
+local function ptu(self, id, pUpdateFunction)
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/commonSmokeApi.lua]: @dboltovskyi done in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/f2c903293464469ebdf036ca148182d0c2bb5501

File: test_scripts/Smoke/commonSmokeApi.lua [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+  self, pAppId = commonSmokeApi.getSelfAndParams(pAppId, self)
+  if not pAppId then pAppId = 1 end
+  return self["mobileSession" .. pAppId]
+end
+
+function commonSmokeApi.splitString(inputStr, sep)
+    if sep == nil then
+        sep = "%s"
+    end
+    local splitted = {}; i = 1
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply test_scripts/Smoke/commonSmokeApi.lua]: @dboltovskyi fixed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1697/commits/6075f7000bab1e04c24699b2e18067a05a0c7126

**Pull request merged**